.github/workflows/scripts/ram_usage.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+# -----------------------------------------------------------------------------------------------------
+# Copyright (c) 2006-2021, Knut Reinert & Freie Universität Berlin
+# Copyright (c) 2016-2021, Knut Reinert & MPI für molekulare Genetik
+# This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+# shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+# -----------------------------------------------------------------------------------------------------
+#
+# Usage ram_usage.py <input_file> <output_file>
+#
+# Computes a table with RAM-Usage from a file containing output of `time -v`.
+import argparse
+import os
+import pandas
+
+parser = argparse.ArgumentParser(description='Compute RAM-Usage', formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+parser.add_argument('input', type=str, help='File containing all outputs of `time -v`.')
+parser.add_argument('output', type=str, help='File to write output to.')
+arguments = parser.parse_args()
+
+file_names = []
+ram_usages = []
+
+with open(arguments.input, 'r') as input_file:
+    parsing_ram_usage = False
+    for line_number, line in enumerate(input_file):
+        if line_number % 23 == 0:
```

Comment:
Can you explain me what this if and the one below (if parsing_ram_usage and ((line_number - 9) % 23) == 0:) are needed for?

Reply:
You have something like this

```
        Command being timed: "/usr/bin/g++-11 [...] -c /home/.../seqan3/test/unit/argument_parser/format_parse_test.cpp"
        User time (seconds): 9.21
        System time (seconds): 0.39
        Percent of CPU this job got: 99%
        Elapsed (wall clock) time (h:mm:ss or m:ss): 0:09.64
        Average shared text size (kbytes): 0
        Average unshared data size (kbytes): 0
        Average stack size (kbytes): 0
        Average total size (kbytes): 0
        Maximum resident set size (kbytes): 672856
        Average resident set size (kbytes): 0
        Major (requiring I/O) page faults: 0
        Minor (reclaiming a frame) page faults: 211593
        Voluntary context switches: 9
        Involuntary context switches: 13
        Swaps: 0
        File system inputs: 8
        File system outputs: 44712
        Socket messages sent: 0
        Socket messages received: 0
        Signals delivered: 0
        Page size (bytes): 4096
        Exit status: 0
        Command being timed: "/usr/bin/g++-11 CMakeFiles/format_parse_test.dir/format_parse_test.cpp.o -o format_parse_test [...]"
        User time (seconds): 0.22
        System time (seconds): 0.04
        Percent of CPU this job got: 76%
        Elapsed (wall clock) time (h:mm:ss or m:ss): 0:00.34
        Average shared text size (kbytes): 0
        Average unshared data size (kbytes): 0
        Average stack size (kbytes): 0
        Average total size (kbytes): 0
        Maximum resident set size (kbytes): 54872
        Average resident set size (kbytes): 0
        Major (requiring I/O) page faults: 0
        Minor (reclaiming a frame) page faults: 15478
        Voluntary context switches: 725
        Involuntary context switches: 0
        Swaps: 0
        File system inputs: 0
        File system outputs: 0
        Socket messages sent: 0
        Socket messages received: 0
        Signals delivered: 0
        Page size (bytes): 4096
        Exit status: 0
```

Every 23 lines there is a new "Command being timed:".
There's the compile step I'm actually interested in which contains the path to the source file (e.g, `seqan3/test/unit/argument_parser/format_parse_test.cpp`.
And then there is the linking step (e.g., `CMakeFiles/format_parse_test.dir/format_parse_test.cpp.o -o format_parse_test`) and it does not contain the path to the source file.

Hence, I can iterate over all the time outputs (blocks of length 23) and check if there is `unit` in the command. If yes, this is something I want the RAM usage of (`parsing_ram_usage`). If I want to parse the RAM usage, and I'm on the 9th line after the beginning of a block, I'm in the line that contains the `Maximum resident set size`.